File: 7. Cyclic rotation of array by one/main .cpp

#include <iostream>

using namespace std;

int main()
{
   int t;
   cin>>t;
   while(t--){
       int n;
       cin>>n;
       
        int arr[n];
        
        for(int i=0;i<n;i++){
            cin>>arr[i];
        }
        int len = sizeof(arr)/sizeof(arr[0]);
        
        int last =arr[len-1];
        for(int i=len-1;i>=0;i--){
            arr[i]=arr[i-1];
            
        }
        
        arr[0]=last;
    
    
        for(int i=0;i<len;i++){
            cout<<arr[i]<<" ";
        }
        cout<<endl;
       
    }
    return 0;
}

File: 7. Cyclic rotation of array by one/main.py
l =[1,2,3,4,5]

last =l[len(l)-1]

temp=l[0]
for i in range(len(l)-1):
    k=l[i+1]
    l[i+1]=temp
    temp=k
    
l[0]=last

print(l)